inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -1365,10 +1433,16 @@ public boolean statementsMatchSPO(KnowledgeBase akb, KBStatement mockStatement)
                     return Optional.of(kbConcept);
                 }
             }
-            else if (!stmts.hasNext()) {
-                Optional<KBInstance> kbInstance = readInstance(aKb, aIdentifier);
-                if (kbInstance.isPresent()) {
-                    return kbInstance.flatMap((p) -> Optional.of(p));
+            else {
+                Optional<KBProperty> kbProperty = readProperty(aKb, aIdentifier);
+                if (kbProperty.isPresent()) {
+                    return kbProperty.flatMap((p) -> Optional.of(p));
```

Comment:
Isn't this a NOP? Can you please add a comment why this is necessary? Same a few lines below.